feat: enhance Jenkins stories's index page with animated gradient background and card component styles

biru-codeastromer opened the pull request:
### Description : 

- This PR introduces changes to `index.jsx` page to be aligned with the "UserStory" page that was brought after the PR #97 .

### Changes : 

- Animated Gradient Background: Introduced a dynamic animated gradient background to give the page a modern and engaging look.

- Consistent Card Component Design: Updated the index page's story card layout to align with the existing styles from userstory page, ensuring design consistency across the site.

- Responsive Design Updates: Ensured the hero section, map size, and cards maintain responsiveness for a consistent user experience across devices.



### Before : 

<img width="1512" alt="Screenshot 2025-02-14 at 7 16 05 PM" src="https://github.com/user-attachments/assets/7a388a36-5525-4fac-9341-ccae5bd14cba" />

------------


### After : 

<img width="1512" alt="Screenshot 2025-02-14 at 7 16 21 PM" src="https://github.com/user-attachments/assets/29309f1d-7595-4c86-a567-e8a2b6b0f10a" />

------------

<img width="1512" alt="Screenshot 2025-02-14 at 7 16 37 PM" src="https://github.com/user-attachments/assets/3ac81583-b84b-4fbc-bfd7-c5ef22326274" />

------------

Please review and let me know for further improvements !

biru-codeastromer commented:
I tried all combos for the page ui to have aligned with the userstory page and at last felt this change nice.

Let me know how this is after review . 




https://github.com/user-attachments/assets/3fe680b6-ae93-4e7a-bedd-caca660783ab

krisstern commented:
I think the new colors are off / not any better than the existing ones and look a bit odd...

biru-codeastromer commented:
> I think the new colors are off / not any better than the existing ones and look a bit odd...

Let me fix that !

krisstern commented:
But shouldn't we at least discuss before you change the colors? Otherwise what if the new colors also don't work? Also, where are you geting the color theme(s)?

biru-codeastromer commented:
> But shouldn't we at least discuss before you change the colors? Otherwise what if the new colors also don't work? Also, where are you geting the color theme(s)?



Sorry for the overlook . I thought you were referring to the white color theme present before in the text .

biru-codeastromer commented:
I will keep in mind this to get confirmed about the intention of the comment more . Sorry for this time . Won't happen again. 


I’d love to hear your thoughts on the best color combo for the homepage. If you have any suggestions or themes in mind, let me know! I’m happy to explore options that work best .

krisstern commented:
Yeah, you may want to show what you have done for the light mode too if any.

biru-codeastromer commented:
> I think you can get color palette ideas from say https://coolors.co/. Their palettes usually have colors that match with each other quite well.



Thank you very much for the reference ! Much appreciated.
 I will look over it clearly and drop some combo ideas to you here to see if it aligns with you too.

biru-codeastromer commented:
> Yeah, you may want to show what you have done for the light mode too if any.

At first , I thought to keep the light mode theme as before but rather found it pale . So for that too I will looking into your reference and give you some combos here for the light theme . Thanks !

biru-codeastromer commented:
Hi, how about this combo in dark theme ?

1. https://coolors.co/palette/0d1b2a-1b263b-415a77-778da9-e0e1dd
<img width="1512" alt="Screenshot 2025-02-14 at 9 14 36 PM" src="https://github.com/user-attachments/assets/5cd51e0b-5b56-4aa9-b3a5-a20c9be9a58a" />


2.https://coolors.co/palette/22223b-4a4e69-9a8c98-c9ada7-f2e9e4
<img width="1512" alt="Screenshot 2025-02-14 at 9 16 32 PM" src="https://github.com/user-attachments/assets/342dcfed-6624-441b-9eea-9c0d3d9bd981" />




What's your view in it ? Please let me know !

krisstern commented:
I prefer the top color palette over the bottom one.

biru-codeastromer commented:
> I prefer the top color palette over the bottom one.

Should I implement and push here the changes for you to confirm ?

krisstern commented:
Yes, please. Next time you don't need to ask again just go ahead and make the code change after I have made the suggestion.

biru-codeastromer commented:
Okay !

biru-codeastromer commented:
I have made the required changes . Let me know if it's looking good now !

biru-codeastromer commented:
> Yeah, you may want to show what you have done for the light mode too if any.


Hi, I have implemented light mode theme too from the colour palette combo . If you don't like any of the theme please let me know I will show you more light mode themes for you to have the preference . Thanks !

krisstern reviewed (changes requested):
The dark mode LGTM.
 
But for the light mode, can we make the text of the highlighted button white in color?:
<img width="1437" alt="Screenshot 2025-02-15 at 2 28 55 AM" src="https://github.com/user-attachments/assets/2c327c34-1194-4c4c-83a5-2f6247712169" />

The contrast is not sufficient otherwise.

biru-codeastromer commented:
Hi, I have implemented your suggestion . Also corrected the button color in dark mode back to original which got changed in my previous commit .

Let me know for further improvements. Thanks !

krisstern reviewed (approved):
LGTM
